cmd/releasego/publish-announcement.go

```diff
@@ -200,23 +198,46 @@ func publishAnnouncement(p subcmd.ParseFunc) (err error) {
 		return fmt.Errorf("file %s already exists in go-devblog repository", blogFilePath)
 	}
 
-	if err := githubutil.Retry(func() error {
-		// Upload the announcement to the go-devblog repositoryy main branch with proper commit message
-		if err := githubutil.UploadFile(
-			ctx,
-			client,
-			org,
-			repo,
-			"main",
-			blogFilePath,
-			fmt.Sprintf("Add new blog post for new release in %s", releaseDate.Format("2006-01-02")),
-			content.Bytes()); err != nil {
-			return fmt.Errorf("error uploading file to go-devblog repository : %w", err)
-		}
+	// Create a feature branch (gitpr convention: dev/<purpose>/<name>) and open a PR to main.
+	prSet := gitpr.PRRefSet{Name: "main", Purpose: fmt.Sprintf("blog-%d", time.Now().Unix())}
+	branchName := prSet.PRBranch()
+
+	if err := githubutil.CreateBranch(ctx, client, org, repo, branchName, "main"); err != nil {
+		return fmt.Errorf("error creating branch %s: %w", branchName, err)
+	}
+
+	if err := githubutil.UploadFile(
+		ctx,
+		client,
+		org,
+		repo,
+		branchName,
+		blogFilePath,
+		fmt.Sprintf("Add blog post: %s", releaseInfo.Title),
+		content.Bytes()); err != nil {
+		return fmt.Errorf("error uploading file to branch %s: %w", branchName, err)
+	}
+
+	// Create PR using gitpr.
+	auther, err := f.Auth.NewAuther()
+	if err != nil {
+		return fmt.Errorf("failed to get GitHub auther: %w", err)
+	}
+
+	ownerRepo := fmt.Sprintf("%s/%s", org, repo)
+	prReq := prSet.CreateGitHubPR(org, releaseInfo.Title,
+		"**Automated Pull Request:** Adds the Microsoft Go release announcement.\nThis PR was generated automatically using the [\"`publish-announcement.go`\"](https://github.com/microsoft/go-infra/blob/main/cmd/releasego/publish-announcement.go) script.")
```

Comment:
IMO this split is clearer:

```suggestion
	prReq := prSet.CreateGitHubPR(
		org,
		releaseInfo.Title,
		"**Automated Pull Request:** Adds the Microsoft Go release announcement.\n"+
			"This PR was generated automatically using the [\"`publish-announcement.go`\"](https://github.com/microsoft/go-infra/blob/main/cmd/releasego/publish-announcement.go) script.")
```